contracts/FragToken.sol

```diff
@@ -108,10 +108,13 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable{
     function unlock(bytes calldata signature) external {
 
         uint256 amount = 0;
+        uint256 len = _lockInfos[msg.sender].length;
+        require(_lockInfos[msg.sender].length > 0, "No previous lock executed");
         // loop over all the locks performed by the sender and calculate the aggregate unlockable
-        for (uint i = _lockInfos[msg.sender].length - 1; i >= 0 ; i--) {
+        for (uint i = len - 1; i >= 0 ; i--) {
             if(_lockInfos[msg.sender][i].locktime < block.timestamp) {
```

Comment:
hey you know what use a reference here
`Lockinfo storage x = _lockInfos[msg.sender]` and use `x` around...

Reply:
Even lines above of course

Reply:
I extracted it and reused. Though I left a comment at this point to explain this case:
```
x[i] = x[x.length - 1]; // `len` here cannot be used because the array is dynamically resized by pop
x.pop();
```

Reply:
> I extracted it and reused. Though I left a comment at this point to explain this case:
> 
> ```
> x[i] = x[x.length - 1]; // `len` here cannot be used because the array is dynamically resized by pop
> x.pop();
> ```

obviously, that's fine, we indeed need recompute here, can't be avoided.